### ISSUE TITLE
Connection was closed when kicking/banning player

### ISSUE THREAD

**Issue by galvand** [haxball/haxball-issues#1151]
is there a way to fix this bug?

**Comment by thenorthstar** [comment]
I suggest you to say your issues more clearly. For example, when this issue is happening? After kicking/banning a player in a normal haxball room; or in a headless bot room? Also where do you kick the players if headless room? On console or in room? It would be helpful for us if you say this more clearly.

**Comment by kingjasko** [comment]
Well, i know from my experience that in headless room (haven't tested it in a normal haxball room) kicks from the bot (with command) often tend to show as connection was closed instead of showing the kicked message

**Comment by thenorthstar** [comment]
> Well, i know from my experience that in headless room (haven't tested it in a normal haxball room) kicks from the bot (with command) often tend to show as connection was closed instead of showing the kicked message

If so, it most probably means that you had missed an argument in **room.kickPlayer()** command. So, the true usage of **room.kickPlayer()** is this:

room.kickPlayer(**id: positive integer**,**message: string**,**boolean: false/true**). Probably you know it now, but there's some interesting thing, if you forget the message argument (**room.kickPlayer(id,false/true()**)), then it kicks the player but the room is crashed. Probably the friend is suffered from this situation now.

**Comment by galvand** [comment]
> I suggest you to say your issues more clearly. For example, when this issue is happening? After kicking/banning a player in a normal haxball room; or in a headless bot room? Also where do you kick the players if headless room? On console or in room? It would be helpful for us if you say this more clearly.

yes . happens in a headless/bot room, and it always happens when the bot tries to kick/ban players as @kingjasko described

**Comment by kingjasko** [comment]
> 
> 
> > Well, i know from my experience that in headless room (haven't tested it in a normal haxball room) kicks from the bot (with command) often tend to show as connection was closed instead of showing the kicked message
> 
> If so, it most probably means that you had missed an argument in **room.kickPlayer()** command. So, the true usage of **room.kickPlayer()** is this:
> 
> room.kickPlayer(**id: positive integer**,**message: string**,**boolean: false/true**). Probably you know it now, but there's some interesting thing, if you forget the message argument (**room.kickPlayer(id,false/true()**)), then it kicks the player but the room is crashed. Probably the friend is suffered from this situation now.

well there you go, i never bothered to look into it, as it wasn't a serious problem for me :P
